src/app/api/run/[runId]/route.ts

```diff
@@ -0,0 +1,47 @@
+import { NextRequest, NextResponse } from 'next/server'
+import { generateAuthorizationHeaders } from '@/app/utils'
+
+const allowedStatusUpdates = ['PROVISIONING', 'RUNNING', 'ERRORED'] as const
```

Comment:
Note for reviewer(s): The `PROVISIONING` state here is speculative and may need to change / get removed depending on final decisions, but that would be a minor tweak.